engine/order_manager.go

```diff
@@ -998,7 +977,9 @@ func (s *store) getByExchange(exchange string) ([]*order.Detail, error) {
 	if !ok {
 		return nil, ErrExchangeNotFound
 	}
-	return r, nil
+	ordersCpy := make([]*order.Detail, len(r))
+	copy(ordersCpy, r)
```

Comment:
This isn't a good way of copying. Please update to loop over orders and run `.Copy()`. 

Example test below
```go
func TestDetailCopySlice(t *testing.T) {
	d := []*Detail{
		{
			Exchange: "Binance",
		},
		{
			Exchange: "Binance",
			Trades: []TradeHistory{
				{Price: 1},
			},
		},
	}

	ordersCpy := make([]*Detail, len(d))
	copy(ordersCpy, d)
	for i := range ordersCpy {
		if !reflect.DeepEqual(d[i], ordersCpy[i]) {
			t.Errorf("[%d] Copy does not contain same elements, expected: %v\ngot:%v", i, d[i], ordersCpy)
		}
		if len(d[i].Trades) > 0 {
			if &d[i].Trades[0] == &ordersCpy[i].Trades[0] {
				t.Errorf("[%d]Trades point to the same data elements", i)
			}
		}
	}
}

```

Reply:
🦅